animation_nodes/nodes/object/lamp_color_input.py

```diff
@@ -0,0 +1,35 @@
+import bpy
+from ... data_structures import Color, ColorList
+from ... base_types import AnimationNode, VectorizedSocket
+
+class LampColorInputNode(bpy.types.Node, AnimationNode):
+    bl_idname = "an_LampColorInputNode"
+    bl_label = "Lamp Color Input"
+    useObjectList: VectorizedSocket.newProperty()
+
+    def create(self):
+        self.newInput(VectorizedSocket("Object", "useObjectList",
+            ("Object", "object", dict(defaultDrawType = "PROPERTY_ONLY")),
+            ("Objects", "objects"),
+            codeProperties = dict(allowListExtension = False)))
+
+        self.newOutput(VectorizedSocket("Color", "useObjectList",
+            ("Color", "color"), ("Colors", "color")))
+
+    def getExecutionFunctionName(self):
+        if self.useObjectList:
+            return "execute_LampColors"
+        else:
+            return "execute_LampColor"
+
+    def execute_LampColor(self, object):
+        if object is None: return Color((0, 0, 0, 0))
+        color = object.data.color
```

Comment:
Should we not check if the object data is actually a lamp?

Reply:
Done.